barman/clients/walrestore.py

```diff
@@ -243,15 +244,14 @@ def try_deliver_from_spool(config, dest_file):
     :param argparse.Namespace config: the configuration from command line
     :param dest_file: The destination file object
     """
-    spool_file = os.path.join(config.spool_dir, config.wal_name)
+    spool_file = Path(config.spool_dir, config.wal_name)
 
     # id the file is not present, give up
-    if not os.path.exists(spool_file):
+    if not spool_file.exists():
         return
 
     try:
-        shutil.copyfileobj(open(spool_file, "rb"), dest_file)
-        os.unlink(spool_file)
+        shutil.move(spool_file, dest_file)
```

Comment:
We discussed this and yes we do agree that this change is a performance improvement in the same fs scenario you described. I'm fine with merging this once we remove the `Pathlib` usage above